bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -215,56 +221,43 @@ std::unique_ptr<ClientReplyMsg> ClientsManager::allocateNewReplyMsgAndWriteToSto
 // * allocate new ClientReplyMsg.
 // * copy reply from reserved pages to ClientReplyMsg.
 // * set primary id.
-std::unique_ptr<ClientReplyMsg> ClientsManager::allocateReplyFromSavedOne(NodeIdType clientId,
+std::shared_ptr<ClientReplyMsg> ClientsManager::allocateReplyFromSavedOne(NodeIdType clientId,
                                                                           ReqId requestSeqNum,
                                                                           uint16_t currentPrimaryId) {
   const uint16_t clientIdx = clientIdToIndex_.at(clientId);
   const uint32_t firstPageId = clientIdx * reservedPagesPerClient_;
   LOG_DEBUG(CL_MNGR, KVLOG(clientId, requestSeqNum, firstPageId));
-  stateTransfer_->loadReservedPage(resPageOffset() + firstPageId, sizeOfReservedPage_, scratchPage_);
-
-  ClientReplyMsgHeader* replyHeader = (ClientReplyMsgHeader*)scratchPage_;
-  ConcordAssert(replyHeader->msgType == MsgCode::ClientReply);
-  ConcordAssert(replyHeader->currentPrimaryId == 0);
-  ConcordAssert(replyHeader->replyLength > 0);
-  ConcordAssert(replyHeader->replyLength + sizeof(ClientReplyMsgHeader) <= maxReplySize_);
-
-  uint32_t replyMsgSize = sizeof(ClientReplyMsgHeader) + replyHeader->replyLength;
-  uint32_t numOfPages = replyMsgSize / sizeOfReservedPage_;
-  uint32_t sizeLastPage = sizeOfReservedPage_;
-  if (replyMsgSize % sizeOfReservedPage_ != 0) {
-    numOfPages++;
-    sizeLastPage = replyMsgSize % sizeOfReservedPage_;
-  }
-  LOG_DEBUG(CL_MNGR, KVLOG(clientId, numOfPages, sizeLastPage));
-  auto r = std::make_unique<ClientReplyMsg>(myId_, replyHeader->replyLength);
-
-  // load reply message from reserved pages
-  for (uint32_t i = 0; i < numOfPages; i++) {
-    char* const ptrPage = r->body() + i * sizeOfReservedPage_;
-    const uint32_t sizePage = ((i < numOfPages - 1) ? sizeOfReservedPage_ : sizeLastPage);
-    stateTransfer_->loadReservedPage(resPageOffset() + firstPageId + i, sizePage, ptrPage);
-  }
+  for (int j = 0; j < maxNumOfReqsPerClient_; j++) {
```

Comment:
Nit: I suggest using the same type as `maxNumOfReqsPerClient_`.

Reply:
Fixed, thanks.